ui/src/app/pages/admin/admin-user-profile.spec.tsx

```diff
@@ -0,0 +1,492 @@
+      VERILY.shortName
+    );
+
+    await simulateComponentChange(
+      wrapper,
+      findDropdown(wrapper, 'verifiedInstitution'),
+      BROAD.shortName
+    );
+    expect(findDropdown(wrapper, 'verifiedInstitution').props.value).toEqual(
+      BROAD.shortName
+    );
+
+    // also need to set the Institutional Role
+
+    await simulateComponentChange(
+      wrapper,
+      findDropdown(wrapper, 'institutionalRole'),
+      InstitutionalRole.POSTDOCTORAL
+    );
+    expect(findDropdown(wrapper, 'institutionalRole').props.value).toEqual(
+      InstitutionalRole.POSTDOCTORAL
+    );
+
+    await simulateTextInputChange(
+      findTextInput(wrapper, 'contactEmail'),
+      BROAD_ADDR_1
+    );
+    expect(findTextInput(wrapper, 'contactEmail').props().value).toEqual(
+      BROAD_ADDR_1
+    );
+
+    expect(wrapper.find('[data-test-id="email-invalid"]').exists()).toBeFalsy();
+
+    const saveButton = wrapper.find('[data-test-id="update-profile"]');
+    expect(saveButton.exists()).toBeTruthy();
+    expect(saveButton.props().disabled).toBeFalsy();
+  });
+
+  it('should prohibit updating institutional role to Other without adding other-text', async () => {
+    const contactEmail = 'user1@google.com';
+    updateTargetProfile({
+      verifiedInstitutionalAffiliation: {
+        ...TARGET_USER_PROFILE.verifiedInstitutionalAffiliation,
+        institutionShortName: VERILY.shortName,
+      },
+      contactEmail,
+    });
+
+    const wrapper = component();
+    expect(wrapper).toBeTruthy();
+    await waitOneTickAndUpdate(wrapper);
+
+    await simulateComponentChange(
+      wrapper,
+      findDropdown(wrapper, 'institutionalRole'),
+      InstitutionalRole.OTHER
+    );
+    expect(findDropdown(wrapper, 'institutionalRole').props.value).toEqual(
+      InstitutionalRole.OTHER
+    );
+
+    let saveButton = wrapper.find('[data-test-id="update-profile"]');
+    expect(saveButton.exists()).toBeTruthy();
+    expect(saveButton.props().disabled).toBeTruthy();
+
+    // now update other-text
+
+    expect(
+      findTextInput(wrapper, 'institutionalRoleOtherText').exists()
+    ).toBeTruthy();
+    const roleDetails = 'I do a science';
+    await simulateTextInputChange(
+      findTextInput(wrapper, 'institutionalRoleOtherText'),
+      roleDetails
+    );
+    expect(
+      findTextInput(wrapper, 'institutionalRoleOtherText').props().value
+    ).toEqual(roleDetails);
+
+    saveButton = wrapper.find('[data-test-id="update-profile"]');
+    expect(saveButton.exists()).toBeTruthy();
+    expect(saveButton.props().disabled).toBeFalsy();
+  });
+
+  it('should allow updating initial credit limit', async () => {
+    const wrapper = component();
+    expect(wrapper).toBeTruthy();
+    await waitOneTickAndUpdate(wrapper);
+
+    expect(
+      findDropdown(wrapper, 'initial-credits-dropdown').props.value
+    ).toEqual(TARGET_USER_PROFILE.freeTierDollarQuota);
+
+    const newLimit = 800.0;
+    await simulateComponentChange(
```

Comment:
I think checking pre-changed select value would be good to add.

Reply:
you mean that I should show this is actually different from the previous?

Reply:
yes.